File: core/src/main/java/com/sequenceiq/cloudbreak/controller/mapper/SpringAccessDeniedExceptionMapper.java
package com.sequenceiq.cloudbreak.controller.mapper;

import javax.ws.rs.core.Response.Status;

import org.springframework.security.access.AccessDeniedException;
import org.springframework.stereotype.Component;

@Component
public class SpringAccessDeniedExceptionMapper extends BaseExceptionMapper<AccessDeniedException> {

    @Override
    Status getResponseStatus() {
        return Status.FORBIDDEN;
    }

    @Override
    Class<AccessDeniedException> getExceptionType() {
        return AccessDeniedException.class;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/mapper/SpringAccessDeniedExceptionMapper.kt
package com.sequenceiq.cloudbreak.controller.mapper

import javax.ws.rs.core.Response
import javax.ws.rs.ext.ExceptionMapper
import javax.ws.rs.ext.Provider

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.security.access.AccessDeniedException

import com.sequenceiq.cloudbreak.controller.json.ExceptionResult
import com.sequenceiq.cloudbreak.logger.MDCBuilder

@Provider
class SpringAccessDeniedExceptionMapper : ExceptionMapper<AccessDeniedException> {

    override fun toResponse(exception: AccessDeniedException): Response {
        MDCBuilder.buildMdcContext()
        LOGGER.error(exception.message, exception)
        return Response.status(Response.Status.FORBIDDEN).entity(ExceptionResult(exception.message)).build()
    }

    companion object {

        private val LOGGER = LoggerFactory.getLogger(SpringAccessDeniedExceptionMapper::class.java)
    }
}
